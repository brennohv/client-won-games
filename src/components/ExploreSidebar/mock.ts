export default [
  {
    title: 'Price',
    name: 'price',
    type: 'checkbox',
    field: [
      {
        name: 'under-$50',
        label: 'Under $50'
      },
      {
        name: 'under-$100',
        label: 'Under $100'
      },
      {
        name: 'under-$150',
        label: 'Under $150'
      },
      {
        name: 'under-$200',
        label: 'Under $200'
      },
      {
        name: 'free',
        label: 'Free'
      },
      {
        name: 'discounted',
        label: 'Dicounted'
      }
    ]
  },
  {
    title: 'Sort by',
    name: 'sort-by',
    type: 'radio',
    field: [
      {
        name: 'high-to-low',
        label: 'High to low'
      },
      {
        name: 'low-to-high',
        label: 'Low to high'
      }
    ]
  },
  {
    title: 'System',
    name: 'system',
    type: 'checkbox',
    field: [
      {
        name: 'windows',
        label: 'Windows'
      },
      {
        name: 'linux',
        label: 'Linux'
      },
      {
        name: 'mac',
        label: 'MAC'
      }
    ]
  },
  {
    title: 'Genre',
    name: 'genre',
    type: 'checkbox',
    field: [
      {
        name: 'action',
        label: 'Action'
      },
      {
        name: 'adventure',
        label: 'Adventure'
      },
      {
        name: 'fps',
        label: 'FPS'
      },
      {
        name: 'mmorpg',
        label: 'MMORPG'
      },
      {
        name: 'rpg',
        label: 'RPG'
      },
      {
        name: 'shooters',
        label: 'Shooters'
      },
      {
        name: 'simulation',
        label: 'Simulation'
      }
    ]
  }
]
