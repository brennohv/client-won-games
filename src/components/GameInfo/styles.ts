import styled, { css } from 'styled-components'
import * as ButtonStyles from 'components/Button/styles'
import * as RibbonStyles from 'components/Ribbon/styles'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    display: flex;
    flex-direction: column;
    background-color: ${theme.colors.white};
    padding: ${theme.spacings.small} ${theme.spacings.small}
      calc(${theme.spacings.xsmall} / 2);

    ${RibbonStyles.Wrapper} {
      padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
      font-size: ${theme.font.sizes.xlarge};
      &::before {
        display: none;
      }
    }

    ${ButtonStyles.Wrapper} {
      font-weight: 500;
    }

    ${media.greaterThan('medium')`
      ${RibbonStyles.Wrapper} {
        right: ${theme.spacings.large};
        top: ${theme.spacings.large};
      `}
    }

    ${media.lessThan('large')`
      ${ButtonStyles.Wrapper} {
        width: 100%
      }
    `}
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.gray};
    margin-bottom: ${theme.spacings.small};

    ${media.greaterThan('medium')`
      max-width: 77rem
    `}
  `}
`

export const WrapperButton = styled.div`
  ${({ theme }) => css`
    align-self: flex-end;
    ${ButtonStyles.Wrapper} {
      margin-bottom: ${theme.spacings.xsmall};
    }
    ${media.greaterThan('medium')`
      display: flex;
      flex-direction: row-reverse;
    `}
  `}
`
