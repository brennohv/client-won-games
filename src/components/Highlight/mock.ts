export default {
  title: 'Read Dead it’s back',
  subtitle: 'Come see John’s new adventures',
  backgroundImage: '/img/background.png',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2'
}
