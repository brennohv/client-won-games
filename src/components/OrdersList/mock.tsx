export default [
  {
    title: 'Red Dead Redemption',
    price: '$215,00',
    img: 'https://source.unsplash.com/user/willianjusten/151x70',
    linkDownload: 'http:teste',
    paymentInfo: {
      flag: 'mastercard',
      img: '/img/master-card.png',
      number: '**** **** **** 4326',
      purchaseDate: 'Purchase made on 07/20/2020 at 20:32'
    }
  },
  {
    title: 'Red Dead Redemption 2',
    price: '$215,00',
    img: 'https://source.unsplash.com/user/willianjusten/151x70',
    linkDownload: 'http:teste',
    paymentInfo: {
      flag: 'mastercard',
      img: '/img/master-card.png',
      number: '**** **** **** 4326',
      purchaseDate: 'Purchase made on 07/20/2020 at 20:32'
    }
  }
]
