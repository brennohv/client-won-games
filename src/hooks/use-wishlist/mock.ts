import {
  MUTATION_UPDATE_WISHLIST,
  MUTATION_CREATE_WISHLIST
} from 'graphql/mutations/wishlist'
import { QUERY_WISHLIST } from 'graphql/queries/wishlist'

export const gameMock = (id: string) => ({
  id,
  cover: {
    url: '/uploads/teste'
  },
  developers: [
    {
      name: 'teste'
    }
  ],
  name: `teste ${id}`,
  price: 50,
  slug: `teste ${id}`,
  __typename: 'Game'
})

export const wishlistMock = {
  //objeto tem que ser igual a requisição e resposta, caso contrario vai retornar um objeto vazio
  request: {
    query: QUERY_WISHLIST,
    context: { session: { jwt: '123' } },
    variables: {
      identifier: 'brennovicentini@gmail.com'
    }
  },
  result: {
    data: {
      wishlists: [
        {
          id: 1,
          games: [gameMock('1'), gameMock('2')]
        }
      ]
    }
  }
}

export const createWishlist = {
  request: {
    query: MUTATION_CREATE_WISHLIST,
    context: { session: { jwt: '123' } },
    variables: {
      input: {
        data: {
          games: ['1']
        }
      }
    }
  },
  result: {
    data: {
      createWishlist: {
        wishlist: {
          id: 1,
          games: [gameMock('1')]
        }
      }
    }
  }
}

export const updateWishlist = {
  request: {
    query: MUTATION_UPDATE_WISHLIST,
    context: { session: { jwt: '123' } },
    variables: {
      input: {
        where: {
          id: 1
        },
        data: {
          games: ['1', '2', '3']
        }
      }
    }
  },
  result: {
    data: {
      updateWishlist: {
        wishlist: {
          id: 1,
          games: [gameMock('1'), gameMock('2'), gameMock('3')]
        }
      }
    }
  }
}

export const wishlistItems = [
  // objeto que vou receber assim que tratar com o wishlistMapper
  {
    id: '1',
    title: 'teste 1',
    slug: 'teste 1',
    img: 'http://localhost:1337/uploads/teste',
    developer: 'teste',
    price: 50
  },
  {
    id: '2',
    title: 'teste 2',
    slug: 'teste 2',
    img: 'http://localhost:1337/uploads/teste',
    developer: 'teste',
    price: 50
  },
  {
    id: '3',
    title: 'teste 3',
    slug: 'teste 3',
    img: 'http://localhost:1337/uploads/teste',
    developer: 'teste',
    price: 50
  }
]
