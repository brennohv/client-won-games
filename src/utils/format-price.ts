export default function formatPrice(price: number | bigint): string {
  return new Intl.NumberFormat('en', {
    style: 'currency',
    currency: 'EUR'
  }).format(price)
}
