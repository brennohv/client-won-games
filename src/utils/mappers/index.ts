import formatPrice from 'utils/format-price'
import { HighlightFragment } from 'graphql/generated/HighlightFragment'
import { QueryGames_games } from 'graphql/generated/QueryGames'
import { QueryHome_banners } from 'graphql/generated/QueryHome'
import { QueryWishlists_wishlists_games } from 'graphql/generated/QueryWishlists'
import { QueryOrders_orders } from 'graphql/generated/QueryOrders'

export const bannerMapper = (banners: QueryHome_banners[]) => {
  return banners.map((banner) => ({
    img: `http://localhost:1337${banner.image?.url}`,
    title: banner.title,
    subtitle: banner.subtitle,
    buttonLabel: banner.button?.label,
    buttonLink: banner.button?.link,
    ...(!!banner.ribbon && {
      ribbon: banner.ribbon.text,
      ribbonSize: banner.ribbon.size,
      ribbonColor: banner.ribbon.color
    })
  }))
}

export const gamesMapper = (games: QueryGames_games[] | undefined | null) => {
  return games
    ? games.map((game) => ({
        id: game.id,
        title: game.name,
        slug: game.slug,
        developer: game.developers[0].name,
        img: `http://localhost:1337${game.cover?.url}`,
        price: game.price
      }))
    : []
}

export const highlightMapper = (
  highlight: HighlightFragment | null | undefined
) => {
  return highlight
    ? {
        title: highlight.title,
        subtitle: highlight.subtitle,
        backgroundImage: `http://localhost:1337${highlight.background?.url}`,
        floatImage: `http://localhost:1337${highlight.floatImage?.url}`,
        buttonLabel: highlight.buttonLabel,
        buttonLink: highlight.buttonLink,
        alignment: highlight.alignment
      }
    : {}
}

export const cartMapper = (games: QueryGames_games[] | undefined) => {
  return games
    ? games?.map((game) => ({
        id: game.id,
        img: `http://localhost:1337${game.cover?.url}`,
        price: formatPrice(game.price),
        title: game.name
      }))
    : []
}

export const wishlistMapper = (
  games: QueryWishlists_wishlists_games[] | undefined
) => {
  return games
    ? games?.map((game) => ({
        id: game.id,
        title: game.name,
        slug: game.slug,
        img: `http://localhost:1337${game.cover?.url}`,
        developer: game.developers[0].name,
        price: game.price
      }))
    : []
}

export const ordersMapper = (orders: QueryOrders_orders[] | undefined) => {
  return orders
    ? orders.map((order) => ({
        flag: order.card_brand ? order.card_brand : null,
        img: order.card_brand
          ? `/public/img/card/${order.card_brand}.png`
          : null,
        number: order.card_last4 ? `**** **** **** ${order.card_last4}` : null,
        purchaseDate: new Intl.DateTimeFormat('en-US', {
          day: 'numeric',
          month: 'short',
          year: 'numeric'
        }).format(new Date(order.created_at)),
        paymentInfo: order.games.map((game) => ({
          id: game.id,
          title: game.name,
          price: formatPrice(game.price),
          img: `${process.env.NEXT_PUBLIC_API_URL}${game.cover?.url}`
        }))
      }))
    : []
}
