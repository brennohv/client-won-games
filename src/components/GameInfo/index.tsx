import Heading from 'components/Heading'
import Button from 'components/Button'
import Ribbon from 'components/Ribbon'
import * as S from './styles'
import {
  AddShoppingCart,
  FavoriteBorder
} from '@styled-icons/material-outlined'

export type GameInfoProps = {
  title: string
  description: string
  price: string
}

const GameInfo = ({ title, description, price }: GameInfoProps) => (
  <S.Wrapper>
    <Heading color="black" lineBottom>
      {title}
    </Heading>

    <Ribbon color="secondary">{`$${price}`}</Ribbon>

    <S.Description>{description}</S.Description>

    <S.WrapperButton>
      <Button size="large" icon={<AddShoppingCart />}>
        Add to cart
      </Button>
      <Button minimal size="large" icon={<FavoriteBorder />}>
        Wishlist
      </Button>
    </S.WrapperButton>
  </S.Wrapper>
)

export default GameInfo
