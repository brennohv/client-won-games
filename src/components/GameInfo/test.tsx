import 'session.mock'
import 'next-image-mock'
import { render, screen } from 'utils/test-utils'

import GameInfo from '.'

const props = {
  id: '1',
  title: 'My title',
  description: 'Game description',
  price: 200
}

describe('<GameInfo />', () => {
  it('should render the game informations', () => {
    const { container } = render(<GameInfo {...props} />)

    // renderizar o heading(title)
    expect(
      screen.getByRole('heading', { name: 'My title' })
    ).toBeInTheDocument()
    // renderizar description
    expect(screen.getByText(props.description)).toBeInTheDocument()
    // renderizar o price
    expect(screen.getByText(/â‚¬200.00/)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render FREE when 0', () => {
    render(<GameInfo {...props} price={0} />)
    expect(screen.getByText(/FREE/)).toBeInTheDocument()
  })
})
