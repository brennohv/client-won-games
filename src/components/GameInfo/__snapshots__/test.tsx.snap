// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfo /> should render the game informations 1`] = `
.c2 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c2::after {
  position: absolute;
  display: block;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient( 180deg,#ff5f5f -14.51%,#f062c0 102.86%,#f23131 102.86% );
  color: #FAFAFA;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 1.6rem 4.8rem;
}

.c8:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c8 svg {
  width: 1.5rem;
}

.c8 svg + span {
  margin-left: 0.8rem;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient( 180deg,#ff5f5f -14.51%,#f062c0 102.86%,#f23131 102.86% );
  color: #FAFAFA;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 1.6rem 4.8rem;
  background: none;
  color: #F231A5;
}

.c10:hover {
  background: none;
}

.c10 svg {
  width: 1.5rem;
}

.c10 svg + span {
  margin-left: 0.8rem;
}

.c10:hover {
  color: #e20e8d;
}

.c4 {
  position: absolute;
  right: 0;
  top: 2.4rem;
  color: #FAFAFA;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3CD3C1;
  font-size: 1.4rem;
  padding: 0.8rem 4.0rem;
  height: 3.4rem;
  right: -1.4rem;
}

.c4::before {
  position: absolute;
  content: '';
  border-style: solid;
  right: 0;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-left-width: 0rem;
}

.c4::before {
  border-top-color: #1f8a7d;
  border-left-color: #1f8a7d;
}

.c4::before {
  top: 3.4rem;
  border-right-width: 1.4rem;
  border-top-width: 0.8rem;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FAFAFA;
  padding: 2.4rem 2.4rem calc(1.6rem / 2);
}

.c0 .c3 {
  padding: 0.8rem 1.6rem;
  font-size: 2.0rem;
}

.c0 .c7 {
  font-weight: 500;
}

.c5 {
  font-size: 1.4rem;
  color: #8F8F8F;
  margin-bottom: 2.4rem;
}

.c6 .c7 {
  margin-bottom: 1.6rem;
}

.c1 {
  width: calc(100% - 7.5rem);
}

.c9 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.8rem;
  }
}

@media (max-width:768px) {
  .c0 .c7 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c5 {
    max-width: 77rem;
  }
}

@media (min-width:768px) {
  .c6 {
    -webkit-align-self: flex-end;
    -ms-flex-item-align: end;
    align-self: flex-end;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
      color="black"
    >
      My title
    </h2>
  </div>
  <div
    class="c3 c4"
    color="secondary"
  >
    â‚¬200.00
  </div>
  <p
    class="c5"
  >
    Game description
  </p>
  <div
    class="c6"
  >
    <button
      class="c7 c8"
    >
      <svg
        aria-hidden="true"
        aria-label="add to cart"
        class="c9"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
        />
      </svg>
      <span>
        Add to cart
      </span>
    </button>
    <button
      class="c7 c10"
    >
      <svg
        aria-hidden="true"
        class="c9"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
      <span>
        Wishlist
      </span>
    </button>
  </div>
</div>
`
